spyder/api/plugins.py

```diff
@@ -132,61 +233,82 @@ def ending_long_process(self, message=""):
             Message to show in the status bar when the long process
             finishes.
         """
-        super(BasePlugin, self)._ending_long_process(message)
+        QApplication.restoreOverrideCursor()
+        self.show_status_message(message, timeout=2000)
+        QApplication.processEvents()
 
+    # TODO: deprecate method and use create_action
+    def register_shortcut(self, qaction_or_qshortcut, context, name,
+                          add_shortcut_to_tip=False):
+        """
+        Register a shortcut associated to a QAction or a QShortcut to
+        Spyder main application.
 
-class SpyderPlugin(BasePlugin):
-    """
-    Spyder plugin class.
+        Parameters
+        ----------
+        qaction_or_qshortcut: QAction or QShortcut
+            QAction to register the shortcut for or QShortcut.
+        context: str
+            Name of the plugin this shortcut applies to. For instance,
+            if you pass 'Editor' as context, the shortcut will only
+            work when the editor is focused.
+            Note: You can use '_' if you want the shortcut to be work
+            for the entire application.
+        name: str
+            Name of the action the shortcut refers to (e.g. 'Debug
+            exit').
+        add_shortcut_to_tip: bool
+            If True, the shortcut is added to the action's tooltip.
+            This is useful if the action is added to a toolbar and
+            users hover it to see what it does.
+        """
+        self.main.register_shortcut(
+            qaction_or_qshortcut,
+            context,
+            name,
+            add_shortcut_to_tip,
+            self.CONF_SECTION,
+        )
 
-    All plugins *must* inherit this class and reimplement its interface.
-    """
-    # ---------------------------- ATTRIBUTES ---------------------------------
+    # --- API: Optional methods to override ----------------------------------
+    # ------------------------------------------------------------------------
+    def get_plugin_title(self):
+        """
+        Return the plugin localized name.
 
-    # Name of the configuration section that's going to be
-    # used to record the plugin's permanent data in Spyder
-    # config system (i.e. in spyder.ini)
-    # Status: Optional
-    CONF_SECTION = None
+        Returns
+        -------
+        str
+            Name of the plugin.
+        """
+        return self.CONF_SECTION
 
-    # One line localized description of the features this plugin implements
-    # Status: Optional
-    DESCRIPTION = None
+    def get_plugin_icon(self):
```

Comment:
Non-dackable plugins can have a configuration page, (like autopep8) and the need an icon for this. Unless we want to defer this to the confpage class, in the case a plugin can define multiple config pages, which is an idea that we may want to explore as well.